spring-cloud-sleuth-core/src/test/java/org/springframework/cloud/sleuth/instrument/web/SkipPatternProviderConfigTest.java

```diff
@@ -97,190 +116,163 @@ public void should_return_empty_when_no_endpoints() {
-				.skipPatternForActuatorEndpointsSamePort(properties,
-						webEndpointProperties, endpointsSupplier)
-				.skipPattern();
-
-		then(patternSamePort).isNotEmpty();
-		then(patternSamePort.get().pattern())
-				.isEqualTo("foo/mgt/(info|info/.*|health|health/.*)");
+		contextRunner.withConfiguration(UserConfigurations.of(EndpointConfig.class))
+				.withPropertyValues("management.endpoint.health.enabled=true",
+						"management.endpoint.info.enabled=true",
+						"management.endpoints.web.exposure.include=info,health",
+						"management.endpoints.web.base-path=/mgt",
+						"server.servlet.context-path=foo")
+				.run(context -> {
+					SkipPatternProvider skipPatternProvider = context
+							.getBean(SkipPatternProvider.class);
+					Pattern pattern = skipPatternProvider.skipPattern();
+					then(pattern.pattern())
+							.contains("foo/mgt/(health|health/.*|info|info/.*)");
+				});
 	}
 
 	@Test
 	public void should_return_endpoints_with_actuator_default_context_path() {
-		WebEndpointProperties webEndpointProperties = new WebEndpointProperties();
-		ServerProperties properties = new ServerProperties();
-		properties.getServlet().setContextPath("/foo");
-		properties.setPort(8080);
-
-		EndpointsSupplier<ExposableWebEndpoint> endpointsSupplier = () -> {
-			ExposableWebEndpoint infoEndpoint = createEndpoint("info");
-			ExposableWebEndpoint healthEndpoint = createEndpoint("health");
-
-			return Arrays.asList(infoEndpoint, healthEndpoint);
-		};
-
-		Optional<Pattern> patternDifferentPort = new TraceWebAutoConfiguration.ActuatorSkipPatternProviderConfig()
-				.skipPatternForActuatorEndpointsDifferentPort(properties,
-						webEndpointProperties, endpointsSupplier)
-				.skipPattern();
+		contextRunner.withConfiguration(UserConfigurations.of(EndpointConfig.class))
+				.withPropertyValues("management.endpoint.health.enabled=true",
+						"management.endpoint.info.enabled=true",
+						"management.endpoints.web.exposure.include=info,health",
+						"server.servlet.context-path=foo")
+				.run(context -> {
+					SkipPatternProvider skipPatternProvider = context
+							.getBean(SkipPatternProvider.class);
+					Pattern pattern = skipPatternProvider.skipPattern();
+					then(pattern.pattern())
+							.contains("foo/actuator/(health|health/.*|info|info/.*)");
+				});
+	}
 
-		then(patternDifferentPort).isNotEmpty();
-		then(patternDifferentPort.get().pattern())
-				.isEqualTo("/actuator/(info|info/.*|health|health/.*)");
+	@Test
+	public void should_return_endpoints_with_actuator_default_context_path_different_port() {
+		contextRunner.withConfiguration(UserConfigurations.of(EndpointConfig.class))
+				.withPropertyValues("management.endpoint.health.enabled=true",
+						"management.endpoint.info.enabled=true",
+						"management.endpoints.web.exposure.include=info,health",
+						"management.servet.port=0", "server.servlet.context-path=foo")
```

Comment:
This is in a few places.

```suggestion
						"management.server.port=0", "server.servlet.context-path=foo")
```

Reply:
Thanks for catching it, I am wondering how come test passes, will check it out

Reply:
Addressed now, the checks are more strict, thanks!